sky/packages/sky/lib/src/widgets/navigator.dart

```diff
@@ -10,13 +10,49 @@ import 'package:sky/src/widgets/transitions.dart';
 
 typedef Widget RouteBuilder(Navigator navigator, RouteBase route);
 
+typedef void NotificationCallback();
+
 abstract class RouteBase {
   Widget build(Navigator navigator, RouteBase route);
   bool get isOpaque;
   void popState([dynamic result]) { assert(result == null); }
-  TransitionBase buildTransition({ Key key });
+
+  AnimationPerformance _performance;
```

Comment:
Originally we were thinking that maybe Navigator and Route should rely on Transitions to manage animations; your change is going in a different direction. I don't have problem with it, just wanted to point it out.